mmif/serialize/mmif.py

```diff
@@ -511,7 +511,57 @@ def get_view_contains(self, at_types: Union[ThingTypesBase, str, List[Union[str,
                 if at_types in view.metadata.contains:
                     return view
         return None
-    
+
+    def _is_in_time_between(self, start: int, end: int, annotation: Annotation) -> bool:
+        s, e = self.get_start(annotation), self.get_end(annotation)
+        if (s < start < e) or (s > start and e < end) or (s < end < e):
+            return True
+        else:
+            return False
+
+    def get_annotations_between_time(self, start: int, end: int) -> Iterator[Annotation]:
+        """
+        Version: 1.0
+        Returns all 'Token' annotations aligned with 'TimeFrame' annotations sorted by start time within start and end time
+        Note: this function only works for mmif object obtained from Whisper-wrapper
+
+        :param start: the start time
+        :param end: the end time
+        :return: a generator of 'Token' annotations
+        """
+        assert start <= end, "Start time must be less than end time"
+        assert start >= 0, "Start time must be greater than or equal to zero"
+        assert end >= 0, "End time must be greater than or equal to zero"
+        # 0. Initialize container and helper method
+        valid_tf_anns = []
+        idtf_to_token = {}
+
+        # 1. find all views that contain the type of TF
+        views = self.get_all_views_contain([AnnotationTypes.TimeFrame, AnnotationTypes.Alignment])
+
+        # 2. For each view, extract annotations that satisfy conditions that are TF/TP and fall into time interval
+        for view in views:
+            tf_anns = view.get_annotations(at_type=AnnotationTypes.TimeFrame)
+            al_anns = view.get_annotations(at_type=AnnotationTypes.Alignment)
+
+            # Select 'TimeFrame' annotations within given time interval
+            for tf in tf_anns:
+                if self._is_in_time_between(start, end, tf):
+                    valid_tf_anns.append(tf)
+
+            # Map 'TimeFrame' id to 'Token' annotation
+            idtf_to_idtk = {align.get_property('source'): align.get_property('target') for align in al_anns}
+            for id_tf in idtf_to_idtk:
+                token_id = idtf_to_idtk[id_tf]
+                idtf_to_token[id_tf] = view.get_annotation_by_id(token_id)
```

Comment:
A few comments on this block 

1. source and target props  of `Alignment` annotation do not have any sense of order (I know, it is pain to deal with such a blend structure), so it can be `TimeFrame` is pointed in the target prop. As long as we can assume the `TF` annotation is in the same view (namely, it is in the `valid_tf_anns`), we can check if either source or target is a (valid) `TF`  and use it as the key, and the other as value. 
2. As mentioned, we only care "valid" time frames, so don't have to iterate entire `idtf`s. 
3. There could be some other types of annotation can be aligned to `TF`s. So the variable name shouldn't limited to "token" (`token` or `tk`, etc)
4. So the above (3) being said, `idtf_to_token` (or its alternative name) might need to be a `Dict[str, List[str]]` to hold (possibly) multiple annotations aligned to the timeframe.

Reply:
I see your point. Now, I changed `idtf_to_token` to map `tf_id` to `List[Annotation]` which represents all aligned annotations of `TimeFrame` whose id is `tf_id`. Using this dict and sorted valid `TimeFrame` annotations, I yielded a "flat" generator by iterating id of each `TimeFrame` within sorted list, because now we know the yielded outcome would be sorted in time as well.